grails-app/controllers/org/pih/warehouse/api/CycleCountApiController.groovy

```diff
@@ -45,21 +48,98 @@ class CycleCountApiController {
         BatchCommandUtils.validateBatch(command, "requests")
         List<CycleCountDto> cycleCounts = cycleCountService.startCycleCount(command)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        withFormat {
+            xls {
+                exportCountXls(cycleCounts)
+            }
+            pdf {
+                renderCountPdf(cycleCounts, command.facility.name)
+            }
```

Comment:
Shouldn't that kind of thing be in the service?

Reply:
No, this should be here, you can check it out in the docs here: https://grails.github.io/legacy-grails-doc/3.2.0/ref/Controllers/withFormat.html